addons/assignGear/functions/fnc_assignGearSupplyBox.sqf

```diff
@@ -40,46 +41,35 @@ if (!isClass _path) exitWith {
-clearWeaponCargoGlobal _theBox;
-clearMagazineCargoGlobal _theBox;
-clearItemCargoGlobal _theBox;
-clearBackpackCargoGlobal _theBox;
-
-private _transportMagazines = getArray(_path >> "TransportMagazines");
-private _transportItems = getArray(_path >> "TransportItems");
-private _transportWeapons = getArray(_path >> "TransportWeapons");
-private _transportBackpacks = getArray(_path >> "TransportBackpacks");
-
-// transportMagazines
-{
-    (_x splitString ":") params ["_classname", ["_amount", "1", [""]]];
-    _theBox addMagazineCargoGlobal [_classname, parseNumber _amount];
-    nil
-} count _transportMagazines; // count used here for speed, make sure nil is above this line
-
-// transportItems
-{
-    (_x splitString ":") params ["_classname", ["_amount", "1", [""]]];
-    _theBox addItemCargoGlobal [_classname, parseNumber _amount];
-    nil
-} count _transportItems; // count used here for speed, make sure nil is above this line
-
-// transportWeapons
-{
-    (_x splitString ":") params ["_classname", ["_amount", "1", [""]]];
-    private _disposableName = [cba_disposable_LoadedLaunchers, _classname, "get", ""] call FUNC(getDisposableInfo);
-    if (_disposableName != "") then {
-        TRACE_2("cba_disposable_LoadedLaunchers replace",_classname,_disposableName);
-        _classname = _disposableName;
+private _subBoxes = "true" configClasses _path;
+if (_subBoxes isNotEqualTo []) then {
+    private _boxName = getText (_path >> "boxCustomName");
+    if (_boxName isNotEqualTo "") then {
+        _theBox setVariable ["ace_cargo_customName", _boxName, true];
     };
-    _theBox addWeaponCargoGlobal [_classname, parseNumber _amount];
-    nil
-} count _transportWeapons; // count used here for speed, make sure nil is above this line
-
-// transportBackpacks
-{
-    (_x splitString ":") params ["_classname", ["_amount", "1", [""]]];
-    _theBox addBackpackCargoGlobal [_classname, parseNumber _amount];
-    nil
-} count _transportBackpacks; // count used here for speed, make sure nil is above this line
+    clearWeaponCargoGlobal _theBox;
+    clearMagazineCargoGlobal _theBox;
+    clearItemCargoGlobal _theBox;
+    clearBackpackCargoGlobal _theBox;
+    private _boxSpace = getNumber (_path >> "boxSpace");
+    [_theBox, [_boxSpace, 4] select (_boxSpace == 0)] call ace_cargo_fnc_setSpace;
+    {
+        private _subBoxType = configName _x;
+        private _boxCount = (getNumber (_x >> "boxCount")) max 1;
+        for "_i" from 1 to _boxCount do {
+            private _subBox = createVehicle [_subBoxType, [23,54,975], [], 0, "CAN_COLLIDE"];
+            [_subBox, _x, ["%1", "%1 " + str _i] select (_boxCount > 1)] call FUNC(setBoxContentsFromConfig);
+            [_subBox, 1] call ace_cargo_fnc_setSize;
+            if !([_subBox, _theBox, true] call ace_cargo_fnc_loadItem) exitWith {
+                diag_log text format ["[POTATO-assignGear] - Failed to create %1 supply box(es) for %2 - out of space ", _subBoxType, typeOf _theBox];
```

Comment:
TIL we don't have any logging macro for this :(

Reply:
This should also be `formatText` instead of `text format`, but I think this bit of code is copy and pasted from your old work